packages/postcss-lowercase-text/src/__tests__/atRules.test.js

```diff
@@ -0,0 +1,120 @@
+import postcss from 'postcss';
+
+import plugin from '..';
+
+function run(testMsg = 'should work', input, output) {
+  it(testMsg, (done) => {
+    let result = postcss([plugin]).process(input, {
+      from: undefined,
+    });
+    expect(result.css).toEqual(output);
+    done();
+  });
+}
+
+describe('transforming @media', () => {
+  run(
+    'should safely transform the @media and its properties',
+    '@MEDIA screen and (min-width: 480px){body{COLOR: lightgreen;}}',
+    '@media screen and (min-width: 480px){body{color: lightgreen;}}'
+  );
+
+  run(
+    'should safely transform the @media but not the params',
+    '@MEDIA SCREEN and (min-width: 480px){}',
+    '@media screen and (min-width: 480px){}'
+  );
+});
+
+describe('transforming @charset', () => {
+  run(
+    'should not transform the @charset',
+    '@charset "UTF-8";',
+    '@charset "UTF-8";'
+  );
+  run(
+    'should safely transform the @charset',
+    '@CHARSET "iso-8859-15";',
+    '@charset "iso-8859-15";'
+  );
+});
+
+describe('transforming @import ', () => {
+  run(
+    'should  transform the @import ',
+    `@IMPORT url("fineprint.css") print;@import url("bluish.css") speech;@ImPoRt 'custom.css';@import url("CHROME://communicator/SKIN/");@import "common.css" screen;`,
+    `@import url("fineprint.css") print;@import url("bluish.css") speech;@import 'custom.css';@import url("CHROME://communicator/SKIN/");@import "common.css" screen;`
+  );
+});
+
+describe('transforming @namespace', () => {
+  run(
+    'should safely transform the @namespace',
+    '@NAMESPACE prefix url(http://www.w3.org/1999/xhtml);',
+    '@namespace prefix url(http://www.w3.org/1999/xhtml);'
+  );
+});
+
+describe('transforming @supports ', () => {
+  run(
+    'should safely transform the @supports ',
+    '@SUPPORTS (display: grid) {div {display: grid;}};@supports not (display: GRID) {div {float: right;}}',
+    '@supports (display: grid) {div {display: grid;}};@supports not (display: GRID) {div {float: right;}}'
+  );
+});
+
+describe('transforming @document  ', () => {
+  run(
+    'should safely transform the @document  ',
+    '@DOCUMENT URL("https://www.EXAMPLE.com/") {h1 {COLOR: green;}}',
+    '@document URL("https://www.EXAMPLE.com/") {h1 {color: green;}}'
```

Comment:
`URL` should be lowercased

Reply:
@evilebottnawi, it will lowercase the argument (`"https://www.EXAMPLE.com/"`) as well which is not saved as lowercasing anything after domain name in URLs are not safe. 
postcss will parse ` URL("https://www.EXAMPLE.com/")` whole as `params` so we cant lowercase only `URL`.

let me know what you think!